src/components/VmActions/index.js

```diff
@@ -106,12 +106,17 @@ const VmActions = ({ vm, isOnCard = false, onGetConsole, onShutdown, onRestart,
     confirmText: 'Suspend?',
     cancelText: 'Cancel',
   }
-
+  let consoleProtocol = ''
+  if (vm.get('consoles').size) {
+    const protocol = (vm.get('consoles').find(c => c.get('protocol') === 'spice') ||
```

Comment:
please extract the outermost expression `(vm.get() ....)` to a separate named variable (`console`?) 
and also consider using `vm.getIn(['consoles', 0])` instead of chained calls to `get`.